The fitP update

balaram26 opened the pull request:
1. I have created the new fitp cpp that has both fitp functions and also multitest functions (this part alone not used for now in python yet).
2. Added some test scripts i used in example section, along with the data used by them.

balaram26 committed:
test prints addeded for running test analysis

bala-hedev committed:
Small changes done

1.fit_p_value cpp modified for contrast data check.
2.division parameter included in compute tuples.
3.Compute max ig discrete function fixed.(had issue with variable initiation).

bala-hedev committed:
Updated package with Working fitP

1. updated the Fitp function
2.Included individual examples scripts for testing MDFS,ComputeIG and fitP value functions.

bala-hedev committed:
Merge branch 'main' of https://github.com/balaram26/mdfs-py

balaram26 committed:
Update .gitignore

Removed some unnecessary lines

yoctozepto commented on mdfs/fit_p_value.py:
```diff
@@ -27,23 +27,25 @@ def __del__(self):
 
 def fit_p_value(chisq, chisq_contrast, *,
-                exponential_fit,
+                dimensions=None,
```
Why does the fit depend on dimensions now? Previously it only needed to know whether it is going to be exponential or not.

balaram26 replied:
Yes its not needed as its same as the bool variable. SO i removed and made the commit based on ur comments now.

yoctozepto commented on mdfs/compute_tuples.py:
```diff
@@ -145,7 +145,7 @@ def compute_tuples(data, decision=None, *, dimensions=2, divisions=1,
 
 
-def compute_tuples_discrete(data, decision=None, *, dimensions=2, pc_xi=0.25,
+def compute_tuples_discrete(data, decision=None, *, divisions = 1, dimensions=2, pc_xi=0.25,
```
No, we don't want to add this here. They must be set from the data.

balaram26 replied:
But the default code to caluclate the divisions will not work in all cases. This is the code that was used before.
`    divisions = len(set(data)) - 1`
For now i have removed the divisions parameter from the function, but then this above block needs to be fixed or modified.

yoctozepto commented on mdfs/compute_max_ig.py:
```diff
@@ -129,8 +142,15 @@ def compute_max_ig_discrete(data, decision, contrast_data=None, dimensions=1, pc
-
+    
     divisions = len(set(data)) - 1
+    # if divisions!= None and divisions <= 0:
```
Code comments are bad. Please remove.

yoctozepto commented on mdfs/compute_max_ig.py:
```diff
@@ -129,8 +142,15 @@ def compute_max_ig_discrete(data, decision, contrast_data=None, dimensions=1, pc
     else:
         n_contrast_variables = 0
-
+    
```
Please don't add random whitespace.

yoctozepto commented on mdfs/compute_max_ig.py:
```diff
@@ -97,22 +97,35 @@ def compute_max_ig(data, decision, contrast_data=None, dimensions=1, divisions=1
+    return x
 
-def compute_max_ig_discrete(data, decision, contrast_data=None, dimensions=1, pc_xi=0.25, return_tuples=False,
+def compute_max_ig_discrete(data, decision, contrast_data=None, dimensions=1, divisions=1, pc_xi=0.25, return_tuples=False,
```
Ditto about divisions - for discrete we must discover them.

balaram26 replied:
Same as mentioned above, lets anyway discuss this this week.

yoctozepto commented on .DS_Store:
This file should be git-ignored.

yoctozepto commented:
Let's talk about it next week.

balaram26 committed:
Made changes based pull review

1. i have removed the dimensions param from fit_p_value
2.I have removed the divisions param from discrete functions
